### PR TITLE
pdpmake: init at 1.4.1

### PR DESCRIPTION
**Pull request by eownerdead**
## Description of changes

Add [pdpmake](https://github.com/rmyorston/pdpmake), a public domain POSIX make.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pd/pdpmake/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  doCheck = true;
+  checkTarget = "test";
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
```
Comment (MikaelFangel) [inline pkgs/by-name/pd/pdpmake/package.nix:20]: The maintainer and mainprogram tags are missing. Please add them (:

Comment (eownerdead) [reply pkgs/by-name/pd/pdpmake/package.nix:20]: Corrected. Would it be better to add `meta.mainProgram` even when `pname` is the same as it?

Comment (MikaelFangel) [reply pkgs/by-name/pd/pdpmake/package.nix:20]: Yes, because relying only on pname is now deprecated as it lead to some bugs earlier on.

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 276732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pdpmake</li>
  </ul>
</details>

- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [ ] `meta.license` fits upstream license
	- [Upstream](https://github.com/rmyorston/pdpmake/blob/d5214d453662c484378dd7254520c2fb27571057/LICENSE) uses the [Unlicense](https://spdx.org/licenses/Unlicense.html)
- [ ] `meta.platforms` is set
	- Set `meta.platforms = lib.platforms.all` while maintaining the already set
	  `meta.badPlatforms` --- see [here](https://nixos.org/manual/nixpkgs/stable/#var-meta-badPlatforms)
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] the list of `phases` is not overridden
- [ ] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
	- N/A: No phase is overridden
- [ ] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
	- N/A: No patch is applied
- [ ] patches that are remotely available are fetched rather than vendored
	- N/A: No patch is applied

# Possible improvements

1. Your addition to the maintainers list should go on a separate commit before
   adding the package. Split the commit into two.
1. Change the license from `publicDomain` to `unlicense`.
1. Set `meta.platforms` to `platforms.all`.

Good job on your first contribution!, @eownerdead.

File: pkgs/by-name/pd/pdpmake/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    homepage = "https://github.com/rmyorston/pdpmake";
+    description = "Public domain POSIX make";
+    license = licenses.publicDomain;
+    maintainers = with maintainers; [ eownerdead ];
+    mainProgram = "pdpmake";
+    badPlatforms = platforms.darwin; # Requires `uimensat`
```
Comment (onemoresuza) [inline pkgs/by-name/pd/pdpmake/package.nix:26]:
```suggestion
    platforms = platforms.all;
    badPlatforms = platforms.darwin; # Requires `uimensat`
```

File: pkgs/by-name/pd/pdpmake/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  doCheck = true;
+  checkTarget = "test";
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/rmyorston/pdpmake";
+    description = "Public domain POSIX make";
+    license = licenses.publicDomain;
```
Comment (onemoresuza) [inline pkgs/by-name/pd/pdpmake/package.nix:23]:
```suggestion
    license = licenses.unlicense;
```

**Review by onemoresuza — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -5437,6 +5437,15 @@
...
   };
+  eownerdead = {
+    name = "EOWNERDEAD";
+    email = "eownerdead@disroot.org";
+    github = "eownerdead";
+    githubId = 141208772;
+    keys = [{
+      fingerprint = "4715 17D6 2495 A273 4DDB  5661 009E 5630 5CA5 4D63";
+    }];
+  };
```
Comment (piegamesde) [inline maintainers/maintainer-list.nix:5448]: Please split this change into a separate commit: "maintainers: add eownerdead"

**Review by piegamesde — DISMISSED** [review]

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 276732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pdpmake</li>
  </ul>
</details>

**Review by onemoresuza — APPROVED** [review]

**Comment by eownerdead** [comment]
Thanks for the kind reviews and help!
And a happy new year!

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 276732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pdpmake</li>
  </ul>
</details>
